### ISSUE TITLE
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined

### ISSUE THREAD

**Issue by gavry** [styxlab/next-cms-ghost#56]
I get this error when trying to deploy directly to Vercel from the ReadMe "Deploy" link:

20:44:38 | $ next-sitemap
-- | --
20:44:38 | Done in 48.01s.
20:44:47 | Traced Next.js serverless functions for external files in: 8665.374ms
20:44:48 | Compressed shared serverless function files: 1757.039ms
20:44:49 | All serverless functions created in: 230.176ms

20:44:49 | TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
-- | --
20:44:49 | at validateString (internal/validators.js:120:11)
20:44:49 | at Object.join (path.js:1039:7)
20:44:49 | at Object.build (/vercel/0a7b738f43024159/.build-utils/.builder/node_modules/@vercelruntimes/next/dist/index.js:1:524138)
20:44:49 | at async buildStep (/var/task/sandbox-worker.js:116502:24)
20:44:49 | at async /var/task/sandbox-worker.js:116184:27 {
20:44:49 | code: 'ERR_INVALID_ARG_TYPE'
-- | --
20:44:49 | }
20:44:51 | Done with "package.json"
-- | --

I originally had it on my own forked version, so I tested it on the original repo. It happens at the end of the build, and doesn't occur in a local build process. Can't figure out what exactly is happening, and didn't find any very similar issues to troubleshoot with. Seems like it may be an issue with the serverless functions being generated?

Thanks so much for the awesome work and any response!

EDIT: Tested it with updated version of next-sitemap in case that was causing an issue; it didn't make any difference.
EDIT2: Tested it without postbuild script of next-sitemap; no dice :(

**Comment by machinelearnear** [comment]
I was having the same issue, and what was causing it was that within my Ghost headless CMS I had uploaded an image that had a space in the name ... i.e. `ML Logo.svg`.

**Comment by gavry** [comment]
Thanks for the reply. Maybe that was it, but I looked through everything from my own Ghost setup and couldn't find any weird filepaths. Then, after changing nothing, I tested the build again today (I tested it like 8 times yesterday) and it worked.

I wonder if it might have been caused by an image file pulled by the repo by default (a social icon or something?) that had a messed up URL reference temporarily. Only reason I think that is cause the same issue happened when I used the deploy button without configuring a CMS.

Anyways, if anyone fixed anything, thanks!
